cmd/nvidia-k8s-vgpu-dm/main.go

```diff
@@ -124,6 +137,113 @@ func (m *SyncableVGPUConfig) Get() string {
 	return m.lastRead
 }
 
+// waitForVFs waits for SR-IOV Virtual Functions to be created on all NVIDIA GPUs.
+// It polls sriov_numvfs until all GPUs have their full VF count enabled.
+func waitForVFs(timeout time.Duration) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	return wait.PollUntilContextCancel(ctx, 5*time.Second, true, func(context.Context) (bool, error) {
+		statuses, err := getNvidiaVFStatus()
+		if err != nil {
+			log.Debugf("Error checking for VFs: %v", err)
+			return false, nil
+		}
+
+		if len(statuses) == 0 {
+			log.Info("No NVIDIA GPUs with SR-IOV capability found, skipping VF wait")
+			return true, nil
+		}
+
+		totalEnabled := 0
+		totalExpected := 0
+		for _, s := range statuses {
+			totalEnabled += s.numVFs
+			totalExpected += s.totalVFs
+		}
+
+		if totalEnabled == totalExpected {
+			log.Infof("All %d VF(s) enabled on %d NVIDIA GPU(s)", totalEnabled, len(statuses))
+			return true, nil
+		}
+
+		log.Infof("Waiting for VFs: %d/%d enabled across %d GPU(s)", totalEnabled, totalExpected, len(statuses))
+		for _, s := range statuses {
+			if s.numVFs < s.totalVFs {
+				log.Debugf("  %s: %d/%d VFs", s.pciAddr, s.numVFs, s.totalVFs)
+			}
+		}
+		return false, nil
+	})
+}
+
+// getNvidiaVFStatus returns SR-IOV status for all NVIDIA GPUs.
+func getNvidiaVFStatus() ([]vfStatus, error) {
+	gpuDevicePaths, err := findNvidiaSRIOVPFs()
+	if err != nil {
+		return nil, err
+	}
+
+	var statuses []vfStatus
+	for _, devicePath := range gpuDevicePaths {
+		numvfs, err := readSysfsInt(filepath.Join(devicePath, "sriov_numvfs"))
+		if err != nil {
+			numvfs = 0
+		}
+		totalvfs, err := readSysfsInt(filepath.Join(devicePath, "sriov_totalvfs"))
+		if err != nil {
+			continue
+		}
+		statuses = append(statuses, vfStatus{
+			pciAddr:  filepath.Base(devicePath),
+			numVFs:   numvfs,
+			totalVFs: totalvfs,
+		})
+	}
+	return statuses, nil
+}
+
+// findNvidiaSRIOVPFs finds all NVIDIA PCI devices that support SR-IOV.
+func findNvidiaSRIOVPFs() ([]string, error) {
```

Comment:
I think SR-IOV is more or less implied when we refer to VFs/PFs
```suggestion
func findNvidiaPFs() ([]string, error) {
```